src/main/java/org/zalando/baigan/proxy/handler/ContextAwareConfigurationMethodInvocationHandler.java

```diff
@@ -76,7 +85,7 @@ private Class<?> getClass(final Object proxy) {
         return interfaces[0];
     }
 
-    private Object getConfig(final String key) {
+    private Object getConfig(final String key, final List<ContextProvider> contextProviders) {
```

Comment:
As this is a list, I'm missing a test case for more than 1 provider to showcase what happens if there is a conflict in their selection (or how precedence on the providers works in case of matches/no matches).

Reply:
Thanks for bringing this @bocytko, I would like to get some suggestions around this if we should limit the number of context provider to `1` or if there is a need to support multiple context providers.

Reply:
What would be the value of supporting multiple ones? (or not)
You used a list for some reason.

Reply:
> You used a list for some reason.

It was just a side effect of array of `args` in the method `Object handleInvocation(Object proxy, Method method, Object[] args)` to simplify logic. 

I modified the code to select only the first context provider when there is more than one, as the current `ContextProvider` is generic enough to support multiple variables. There is a warning message if there are more than one context provider used as method parameter.

Reply:
Actually, I would allow multiple context providers. If multiple contexts are required, it could be impractical to add them all to the same context provider. For conflicts, you have the options

* fail on getting config
* specify behavior (e.g. take first / last)